### PR TITLE
[Baselines] Use Numpy for CPU Batching

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

Pytorch indexing is slow because it needs to track things for autograd even in no_grad mode, thereby making numpy faster for indexing and batching of obs.

## How Has This Been Tested

With a little test script.  This is about 2x faster sensors with normal Depth and GPS+Compass and becomes more beneficial as more little sensors are added.

## Types of changes

- Docs change / refactoring / dependency upgrade

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: habitat_baselines/utils/common.py [hunk 1]
```diff
@@ -171,21 +173,42 @@ def batch_obs(
...
+                        torch.as_tensor(sensor),
+                        device,
                     )
 
-                batch_t[sensor_name][i].copy_(sensor)
+                # Use isinstance(sensor, np.ndarray) here instead of
+                # np.asarray as this is quickier for the more common
+                # path of sensor being an np.ndarray
+                # np.asarray is ~3x slower than checking
+                if isinstance(sensor, np.ndarray):
```
Comment (Skylion007) [inline habitat_baselines/utils/common.py:192]: @erikwijmans I think I found a better solution here: np.asanyarray https://stackoverflow.com/a/56805220/2444240

Comment (erikwijmans) [reply habitat_baselines/utils/common.py:192]: That's fixing a different problem, `np.asanyarray` slower than `np.asarray` by a very small but still measurable amount in my testing.

Comment (Skylion007) [reply habitat_baselines/utils/common.py:192]: Still I think we should replace the asarray at the bottom with asanyarray.

Comment (erikwijmans) [reply habitat_baselines/utils/common.py:192]: Why?  asanyarray just lets subclasses of `np.ndarray` pass through, but we know sensor cannot be a subclass of `np.ndarray` if that if `isinstance(sensor, np.ndarray)` returned false.

**Comment by Skylion007** [comment]
Oh yeah, good point. NVM.

**Pull request merged**